### PR TITLE
rename key terms across source code and docs

### LINKED ISSUE #232: Renaming key terms
Some specific terms have recently caused some confusion for developers. Propose updating as follows:

- `observations` --> `inputs`
- `impacts` --> `outputs`
- `rimpl` --> `impact`
`calculate` --> `execute` (in the context of the ImpactModel interface)

@jawache @gnanakeethan @narekhovhannisyan

### PR DESCRIPTION
**Pull request by jmcook1186**
### Types of changes

- [ ] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

- `observations` --> `inputs`
- `impacts --> `outputs`
- `rimpl` --> `impact`
- `calculate` --> `execute` (in the context of the ImpactModel interface)

Closes #232

### CODE REVIEW

File: scripts/impact.ts [hunk 1]
```diff
@@ -9,9 +9,9 @@ import {openYamlFileAsObject, saveYamlFileAs} from '../src/util/yaml';
  * 3. Initializes models.
  * 4. Initializes graph, does computing.
  * 5. Saves processed object as a yaml file.
- * @example run following command `npx ts-node scripts/rimpl.ts --impl ./test.yml --ompl ./result.yml`
+ * @example run following command `npx ts-node scripts/impact.ts --impl ./test.yml --ompl ./result.yml`
  */
-const rimplScript = async () => {
+const outputScript = async () => {
```
Comment (narekhovhannisyan) [inline scripts/impact.ts:14]: impact script

File: src/__mocks__/model-universe/index.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
-import {IImpactModelInterface} from '../../lib';
+import {IoutputModelInterface} from '../../lib';
 
 import {
   ImplInitializeModel,
   InitalizedModels,
 } from '../../types/models-universe';
 
-class MockModel implements IImpactModelInterface {
+class MockModel implements IoutputModelInterface {
```
Comment (narekhovhannisyan) [inline src/__mocks__/model-universe/index.ts:8]: `IOutputModelInterface`

File: examples/impls/case-studies/msft-green-ai.yaml [hunk 3]
```diff
@@ -52,7 +52,7 @@ graph:
           functional-unit-duration: 1 
           functional-duration-time: ''
           functional-unit: requests # factor to convert per time to per f.unit
-    observations: 
+    inputs: 
       - timestamp: 2023-07-06T00:00
         duration: 1205 # 22.27 min runtime in seconds
         energy: 14.37
```
Comment (narekhovhannisyan) [inline examples/impls/case-studies/msft-green-ai.yaml:58]: missing newline

**Comment by jmcook1186** [comment]
Thanks for your work on this @narekhovhannisyan - lgtm and all unit and integration tests passing. Since its my PR I can't give it an approving review - can you approve and I'll merge?

**Review by narekhovhannisyan — APPROVED** [review]

**Pull request merged**